src/handlers/chainSpace.ts

```diff
@@ -0,0 +1,44 @@
+import { SubstrateExtrinsic } from "@subql/types";
+import { ChainSpace, Call } from "../types";
+
+export async function createChainSpace(
+  extrinsic: SubstrateExtrinsic,
+  call: Call,
+  id: string,
+  method: string
+): Promise<void> {
+  const data = extrinsic.extrinsic.method;
+
+  // logger.info("\n\n ChainSpace");
+  // logger.info("\n Data Event:");
+  // logger.info(extrinsic.events.toString());
+  // logger.info("\n\n ChainSpace ENDED");
+
+  let arrayed = JSON.parse(JSON.stringify(data));
+  // logger.info(
+  //   `\n\n\n\n ChainSpace Method: ${method} \n ${JSON.stringify(arrayed)}`
+  // );
+  let chainSpaceId = JSON.parse(JSON.stringify(extrinsic.events))[0].event
+    .data[0];
+  let chainspace = new ChainSpace(id);
+
+  if (method === "create") {
+    chainspace.method = method;
+    chainspace.blockNumber = arrayed.args.did_call.blockNumber;
```

Comment:
When retrieving the block number from the arguments, the obtained value corresponds to the block number at the time of transaction creation. It is imperative to capture the block number at the moment when the transaction is anchored on the blockchain to accurately reflect the transaction's finalized state within the chain.